### ISSUE TITLE
How to implement a Tensorflow model in tiny-dnn?

### ISSUE THREAD

**Issue by FohlenAFK** [tiny-dnn/tiny-dnn#964]
Hi guy's,
I have written a simpel network with tensorflow in python and try to get the same result with tiny-dnn in c++ but I'm having trouble implementing the model correctly

  // features = 0-4, label = 5
  // Example Data: 23.18, 27.272, 426, 721.25, 0.00479, 1

```
#!/usr/bin/env python

import tensorflow as tf
import numpy as np
import input_data

def init_weights(shape, dev):
    return tf.Variable(tf.random_normal(shape, stddev=dev))

def model(X, w_h, w_o):
    h = tf.nn.sigmoid(tf.matmul(X, w_h)) # this is a basic mlp, think 2 stacked logistic regressions
    return tf.matmul(h, w_o) # note that we dont take the softmax at the end because our cost fn does that for us

data = input_data.read_data_sets("datasets/");
trX, trY = data.train.features, data.train.labels;
tvX, tvY = data.validate.features, data.validate.labels;
teX, teY = data.test.features, data.test.labels;

X = tf.placeholder("float32", [None, 5]) # features
Y = tf.placeholder("float32", [None, 1]) # labels

w_h = init_weights([5, 100], np.sqrt(2 / np.prod(X.get_shape().as_list()[1:]))) # create symbolic variables
w_o = init_weights([100, 1], np.sqrt(2 / np.prod(Y.get_shape().as_list()[1:])))

py_x = model(X, w_h, w_o)

cost = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits=py_x,labels=Y)) # compute costs
train_op = tf.train.GradientDescentOptimizer(0.05).minimize(cost) # construct an optimizer
predict_op = tf.nn.sigmoid(py_x);

cost_history = [];
accuracy_history = [];

# Launch the graph in a session
with tf.Session() as sess:
    # you need to initialize all variables
    tf.initialize_all_variables().run()

    for i in range(300):
        for start, end in zip(range(0, len(trX), 128), range(128, len(trX)+1, 128)):
            sess.run(train_op, feed_dict={X: trX[start:end], Y: trY[start:end]});

        cost_history.append(sess.run(cost, feed_dict={X: tvX, Y:tvY}));
        print(i, 'cost:', sess.run(cost, feed_dict={X: tvX, Y:tvY}));
        print(i, 'accuracy:', (np.round(sess.run(predict_op, feed_dict={X: tvX})) == tvY).mean())


    print('Final score:', (np.round(sess.run(predict_op, feed_dict={X: teX})) == teY).mean())
```
Final score: 97,5%

```
int main(int argc, char **argv) {
  DataSet trainDataSet = loadData("datasets/data_training.csv");
  DataSet testDataSet = loadData("datasets/data_test.csv");
  
  network<sequential> nn;
  nn << fully_connected_layer(5, 100) << sigmoid_layer()  // features
     << fully_connected_layer(100, 1) << sigmoid_layer(); // labels 

  size_t batch_size = 1;
  size_t epochs = 300;
  gradient_descent optimizer;
  timer t;
  nn.fit<mse>(optimizer, trainDataSet.vInputData, trainDataSet.vOutputData, batch_size, epochs,
              [&](){
                t.elapsed();
                t.reset();
              },
              [&]() {
                result res = nn.test(testDataSet.vInputData,
                                     testDataSet.vOutputDataLabel);
                cout << res.num_success << "/" << res.num_total << endl;
              });

  double loss =
      nn.get_loss<mse>(testDataSet.vInputData, testDataSet.vOutputData);
  cout << "mse : " << loss << endl;

  return 0;
}
```

Result for each epoch: still 78,9889253%

**Comment by honesteecode** [comment]
I think <mse> should change to <cross_entroy>.

**Comment by FohlenAFK** [comment]
Thanks @honesteecode,

I have changed the network to optimize the cross entropy and edit the learning rate:
```
optimizer.alpha = 0.05; // learning rate
nn.fit<cross_entropy>optimizer, trainDataSet.vInputData, trainDataSet.vOutputData, batch_size, epochs)
```

My `trainDataSet.vInputData` has 9500 samples each with 5 values and the 
belonging `trainDataSet.vOutputData` 9500 each only one value with 0 or 1. The same sizes for the testDataSet but with different values.

But the neuronal network does still not optimize, the result of each epoch is still:
1: 7451/9500
2: 7451/9500
3: 7451/9500
...

The result of the `nn.predict(testDataSet.vvtInputData[0]);` should get 0 or 1 but it doesn't.

Do you have another idea?
